grails-app/views/dashboard/megamenu.gsp

```diff
@@ -99,14 +99,43 @@
 
 <script type="text/javascript">
   $(document).ready(function () {
-      const path = window.location.pathname
-      const menuConfigValues = $(".menu-config-value").toArray();
-      const matchingSection = menuConfigValues.find(it => it.value.includes(path))
+    function getHrefs(value) {
```

Comment:
this function seems a little too complicated which makes it very hard to read and understand what is going on.
I think that we should make it more explicit and straight forward. 
My proposition on how to deal with certain pages that are not underlined in the menu for GSP pages is to add a script in eg. show.gsp for stocklists
```
<script type="text/javascript">
  $(document).ready(function () {
    const matchingMenuSection = $("#requisitionTemplate')
    const matchingMenuSectionCollapsable = $("#requisitionTemplate-collapsed").get(0);
    if (matchingMenuSection) matchingMenuSection.classList.add('active-section');
    if (matchingMenuSectionCollapsable) matchingMenuSectionCollapsable.classList.add('active-section');
  })
</script>
``` 
that way when this show page is going to be rendered the script is going to add an active class to the appropriate section

of course you would have to implement this script in all of the pages that require it but I don't think this will be a big problem since it's a simple logic and I would prefer some duplicate chunks of  code instead of a complicated hard to read function